Qpf/Macro/Data/Constructors.lean

```diff
@@ -13,11 +13,7 @@ partial def countConstructorArgs : Syntax → Nat
   | Syntax.node _ ``Term.arrow #[_, _, tail]  =>  1 + (countConstructorArgs tail)
   | _                                         => 0
 
-open Elab
-/--
-  Add convenient constructor functions to the environment
--/
-def mkConstructors (view : DataView) (shape : Name) : CommandElabM Unit := do
+def mkConstructorsWithNameAndType (view : DataView) (shape : Name) (nameGen : CtorView → Ident) (ty : Term) := do
```

Comment:
I'd prefer the return type (presumably `CommandElabM Unit`?) to be explicit.
Also, please write a docstring to explain what this docs (and crucially, how it differs from `mkConstructors`).

Finally, I like to limit lines to 100 chars wide, just like in Mathlib. This line feels like it might be too long (but I haven't checked, we really should set up a linter of some sort)